packages/rulesets/src/spectral/functions/Extensions/long-running-response-status-code.ts

```diff
@@ -1,50 +1,76 @@
 // An x-ms-long-running-operation extension passes this rule if the operation that this extension has a valid response defined.
 
-const longRunningResponseStatusCode = (methodOp: any, _opts: any, ctx: any, validResponseCodesList: any) => {
-  if (methodOp === null || typeof methodOp !== "object") {
+// operations that this rule evaluates
+const RULE_OPERATION_NAMES = ["delete", "put", "patch", "post"]
+// extension to indicate a long-running operation (LRO)
+const LRO_EXTENSION = "x-ms-long-running-operation"
+// valid response codes for long-running (LRO) ARM (Control Plane) operations
+const ARM_LRO_RESPONSE_CODES_BY_OPERATION: any = {
+  delete: ["200", "204"],
+  post: ["200", "201", "202", "204"],
+  put: ["200", "201"],
+  patch: ["200", "201", "202"],
```

Comment:
>201 [](http://example.com/codeflow?start=18&length=3)

no 201 for patch

Reply:
I seem to recall I've queried people about whether PATCH creating resources makes sense before and the consensus seemed to be no... it should be used only for updates. I already forgot why though, sigh.